feat: add Update ComfyUI option to Help Center for non-desktop environments

Myestery opened the pull request:
## Summary

- Adds "Update ComfyUI" menu item to Help Center for portable/localhost environments
- Wires existing `/v2/manager/queue/update_comfyui` endpoint to the frontend
- Only visible in non-desktop, non-cloud distributions (where Electron update mechanism isn't available)

## Changes

1. **Service layer**: Added `updateComfyUI()` method to `comfyManagerService.ts`
2. **UI**: Added menu item with download icon to `HelpCenterMenuContent.vue`
3. **i18n**: Added translation key for the new menu item

## Context

The new Manager UI (v4) lost the ability to update ComfyUI core in non-desktop environments. This restores that functionality by integrating the existing manager endpoint into the Help Center menu.

## Test plan

- [ ] Verify menu item appears in portable/localhost environments
- [ ] Verify menu item does NOT appear in desktop (Electron) environments
- [ ] Verify menu item does NOT appear in cloud environments
- [ ] Test clicking the menu item triggers update and reboot

┆Issue is synchronized with this [Notion page](https://www.notion.so/PR-7578-feat-add-Update-ComfyUI-option-to-Help-Center-for-non-desktop-environments-2cc6d73d3650811e9e4fe55515f50333) by [Unito](https://www.unito.io)

Myestery committed:
feat(manager): add updateComfyUI service method

Add updateComfyUI() to comfyManagerService for calling the
/v2/manager/queue/update_comfyui endpoint. This enables updating
ComfyUI core from the frontend in non-desktop environments.

Myestery committed:
feat(help-center): add Update ComfyUI menu item

Add "Update ComfyUI" option to the Help Center menu for non-desktop,
non-cloud environments (portable/localhost). Uses a download icon
and triggers the manager's updateComfyUI endpoint followed by reboot.

- Fixes #issue-number

Myestery committed:
test: add updateComfyUI mock to manager store tests

Update mock to include the new updateComfyUI method.

Myestery committed:
feat(help-center): add toast notifications for ComfyUI update

Show toast notifications when:
- Update has started (info)
- Update completed successfully (success)
- Update failed (error with details)

Myestery committed:
fix(help-center): improve error handling for ComfyUI update

Address CodeRabbit review feedback:
- Add try-catch for exception handling
- Change error condition from && to || for robust null check
- Add fallback error message translation key

Myestery commented on src/components/helpcenter/HelpCenterMenuContent.vue:
```diff
@@ -545,6 +558,13 @@ const onReinstall = (): void => {
+  void updateComfyUI({ is_stable: true }).then(() => {
+    void rebootComfyUI()
+  })
+}
```
Fixed in commit b56b9710d - this was actually already addressed in 2b786dc50 with toast notifications, but the later commit improved the error handling further.

Myestery commented on src/locales/en/main.json:
```diff
@@ -773,7 +773,12 @@
+    "updateComfyUIStarted": "Update Started",
+    "updateComfyUIStartedDetail": "ComfyUI update has been queued. Please wait...",
+    "updateComfyUISuccess": "Update Complete",
+    "updateComfyUISuccessDetail": "ComfyUI has been updated. Rebooting..."
```
Added `updateComfyUIFailed` translation key in commit b56b9710d.

christian-byrne reviewed (commented):
Nice. Just need to check useManagerState. If manager is disabled, the service won't work

Myestery committed:
fix(help-center): only show Update ComfyUI when manager is enabled

Check isNewManagerUI from useManagerState before showing the menu item.
The service requires the new manager UI to be active.

Myestery committed:
refactor: move useManagerState call to top level of setup
